lib/hooks/useExternalErc20Awards.js

```diff
@@ -35,53 +32,42 @@ export const fetchErc20AwardBalances = async (
   setErc20Awards
 ) => {
   const batchCalls = []
+  const flattenedValues = []
+
   let etherplexTokenContract
   const { prizePool } = poolAddresses
 
   if (!erc20Addresses || erc20Addresses?.length === 0) {
     return
   }
 
-  try {
-    // Prepare batched calls
-    for (const address of erc20Addresses) {
-      etherplexTokenContract = contract(address, ERC20Abi, address)
-      batchCalls.push(
+  for (const address of erc20Addresses) {
+    etherplexTokenContract = contract(address, ERC20Abi, address)
+    try {
+      const response = await batch(
+        provider,
         etherplexTokenContract
           .balanceOf(prizePool)
           .name()
           .symbol()
           .decimals()
       )
-    }
-
-    const values = await batch(provider, ...batchCalls)
-    const flattenedValues = []
-
-    for (const [address, value] of Object.entries(values)) {
-      const formattedBalance = displayAmountInEther(value.balanceOf[0], {
+      const values = response[address]
+      const formattedBalance = displayAmountInEther(values.balanceOf[0], {
         precision: 4,
-        decimals: value.decimals[0]
+        decimals: values.decimals[0]
```

Comment:
It would be ideal to change these lines to `values.decimals?.[0] || DEFAULT_TOKEN_PRECISION` and set `export const DEFAULT_TOKEN_PRECISION = 18` in the constants file